lib/src/credentials.dart

```diff
@@ -61,9 +62,13 @@ enum AuthProviderType {
 
 extension AuthProviderTypeInternal on AuthProviderType {
   static AuthProviderType getByValue(int value) {
-    for (final type in AuthProviderType.values) {
-      if (type._value == value) {
-        return type;
+    if (value == AuthProviderType._anonymousNoReuse._value) {
+      return AuthProviderType.anonymous;
+    } else {
+      for (final type in AuthProviderType.values) {
+        if (type._value == value) {
+          return type;
+        }
       }
     }
```

Comment:
```suggestion
    if (value == AuthProviderType._anonymousNoReuse._value) {
      return AuthProviderType.anonymous;
    }
    
    return AuthProviderType.values.firstWhere((v) => v._value == value);
```

And we can remove the argument exception at the end.